src/site/content/en/blog/optimize-long-tasks/index.md

```diff
@@ -0,0 +1,399 @@
+{% endAside %}
+
+The result is that the once-monolithic task is now broken up into separate tasks.
+
+<figure>
+  {% Img src="image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/wg0FW6S29CzOCbbwk9kK.png", alt="The same saveSettings function depicted in Chrome's performance profiler, only with yielding. The result is the once-monolithic task is now broken up into five separate tasks&mdash;one for each function.", width="800", height="211" %}
+  <figcaption>
+    The <code>saveSettings</code> function now executes its child functions as separate tasks.
+  </figcaption>
+</figure>
+
+The benefit of using a promise-based approach to yielding rather than manual use of `setTimeout` is better ergonomics. Yield points become declarative, and therefore easier to write, read, and understand.
+
+### Yield only when necessary
+
+What if you have a bunch of tasks, but you only want to yield if the user attempts to interact with the page? That's the kind of thing that [`isInputPending`](/isinputpending/) was made for!
+
+`isInputPending` is a function you can run at any time to determine if the user is attempting to interact with a page element: a call to `isInputPending` will return `true`. It returns `false` otherwise.
+
+Say you have a queue of tasks you need to run, but you don't want to get in the way of any inputs, such as an attempt to open a navigation menu, for example. This code which uses both `isInputPending` and our custom `yieldToMain()` function ensures that the input won't be delayed while the user is trying to interact with the page:
+
+```js
+async function saveSettings () {
+  // A task queue of functions
+  const tasks = [
+    validateForm,
+    showSpinner,
+    saveToDatabase,
+    updateUI,
+    sendAnalytics
+  ];
+  
+  while (tasks.length > 0) {
+    // Yield to a pending user input:
+    if (navigator.scheduling.isInputPending()) {
+      // There's a pending user input. Yield here:
+      await yieldToMain();
+    } else {
+      // Shift the the task out of the queue:
+      const task = tasks.shift();
+
+      // Run the task:
+      task();
+    }
+  }
+}
+```
+
+While `saveSettings` runs, it will loop over the tasks in the queue. If `isInputPending` ever returns `true` during the loop, `saveSettings` will call `yield` so the user input can be handled. Otherwise, it will shift the next task off the front of the queue and run it continuously. It will do this until no more tasks are left.
+
+<figure>
+  {% Img src="image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/K0PEngJQFDDOxn9kM8ne.png", alt="A depiction of the saveSettings function running in Chrome's performance profiler. The resulting task blocks the main thread until isInputPending returns true, in which case, the task yields to the main thread.", width="800", height="254" %}
```

Comment:
This image doesn't look quite right to me. The functions should all be run as part of a single task until the code yields, which shouldn't happen until after input. In this image I can see several distinct tasks before the input occurs.

I get that this particular example is a bit hard to repro, but I was able to kinda get something that works here: 

<img width="1904" alt="Screen Shot 2022-09-29 at 7 21 51 PM" src="https://user-images.githubusercontent.com/326742/193177949-0b815592-69ec-46a4-bfe0-90aac0460f0f.png">

To do this I made a [fork of your glitch](https://task-chunking-demo2.glitch.me/) and tweaked the code a bit (probably easier to explain over GVC if that's helpful).

Either way, what I like about this particular screenshot is it shows a long task at the beginning, but despite the long task the interaction is still only 24ms because the code yields before starting the next task.

WDYT?

Reply:
This works great, but I'd like to get a screenshot of this without the hover—and it's almost impossible for me to repro because I can't get my cursor on the "toggle box" bit fast enough. I'll have to keep trying until I get it.

Reply:
Nailed it. Check the updated figure in the latest commit and let me know what you think.